packages/graphql-lowcode/src/generate/roots/roots.ts

```diff
@@ -0,0 +1,47 @@
+import { IntrospectionQuery, TypesObject, Root } from '../types'
+
+/**
+ * @param introspection Introspection JSON `data`
+ * @returns Query, mutation and subscription root names or undefined if no root is defined.
+ */
+
+export function getRootNames(introspection: IntrospectionQuery): { type: string, name?: string }[] {
+  const queryTypeName = introspection.queryType?.name
+  const mutationTypeName = introspection.mutationType?.name
+  const subscriptionTypeName = introspection.subscriptionType?.name
+
+  return [
+    { type: 'query', name: queryTypeName },
+    { type: 'mutation', name: mutationTypeName },
+    { type: 'subscription', name: subscriptionTypeName }
+  ]
+}
+
+/**
+ * 
+ * @param types Introspection JSON `data.types`
+ * @param queryRootName Query root name
+ * @param mutationRootName Mutation root name
+ * @param subscriptionRootName Subscription root name
+ * @returns Array of root type objects `[query, mutation, subscription]`
+ */
+
+export function getRoots(
+  types: TypesObject[],
+  rootNames: { type: string, name?: string }[],
+): (Root | undefined)[] {
+  let roots: (Root | undefined)[] = []
+
+  rootNames.forEach(root => {
```

Comment:
check every loop you used if possible to use array.map(), array.find(), array.filter()...